src/ingest-pipeline/misc/tools/zarr_zipper.py

```diff
@@ -0,0 +1,130 @@
+        if UUID_REGEX.match(str(elt)):
+            uuid = str(elt)
+            break
+    else:
+        raise RuntimeError("no UUID found")
+    return target_path, str(path.name), uuid
+
+
+def run_zip(target_path: Path, zarr_name: str) -> None:
+    """
+    Zips the directory appropriately for use as a zipped zarr, and deletes
+    the original.
+
+    cd '/hive/hubmap-dev/data/consortium/IEC Testing Group/672111342c0002f90e1cb1bc5686ecf1/anndata-zarr/reg001_expr-anndata.zarr'
+    zip -r ../reg001_expr-anndata.zarr.zip .
+    cd ..
+    mv reg001_expr-anndata.zarr ../../672111342c0002f90e1cb1bc5686ecf1_anndata-zarr_reg001_expr-anndata.zarr
+    """
+    full_target_path = target_path / zarr_name
+    zip_name = target_path / f"{zarr_name}.zip"
+    cmd1 = f"cd '{full_target_path}'"
+    cmd2 = f"zip -r '{zip_name}' ."
+    cmd3 = "cd .."
+    cmd4 = f"rm -r -v {zarr_name}"
+    full_cmd = f"{cmd1} && {cmd2} && {cmd3} && {cmd4}"
+    run(full_cmd, shell=True, check=True)
+
+
+def restructure(candidate: Path, dryrun: bool = False) -> str:
+    try:
+        target_path, zarr_name, uuid = parse_path(candidate)
+    except RuntimeError as excp:
+        LOGGER.warn(f"error parsing {candidate}: {excp}")
+        return None
+    LOGGER.info(f"<{target_path}> <{zarr_name}> <{uuid}>")
+    if dryrun:
+        LOGGER.info(f"dryrun is true; not zipping {zarr_name}")
+    else:
+        run_zip(target_path, zarr_name)
+    return uuid
+
+
+def verify_uuid(uuid: str) -> bool:
+    """
+    Take ENTITY_API and AUTH_TOK from the environment and look up the given uuid.
+    Returns True if a valid result is found, False otherwise.
+    """
+    auth_tok = os.environ.get("AUTH_TOK")
+    if not auth_tok:
+        raise RuntimeError("AUTH_TOK is not defined in the environment")
+    entity_api = os.environ.get("ENTITY_API")
+    if not entity_api:
+        raise RuntimeError("ENTITY_API is not defined in the environment")
+    resp = requests.get(
+        url=f"{entity_api}/entities/{uuid}",
+        headers={
+            "Authorization": f"Bearer {auth_tok}"
+        }
+    )
+    try:
+        resp.raise_for_status()
+    except Exception as excp:
+        LOGGER.error(f"Validation of {uuid} failed: {type(excp)} {excp}")
+        return False
+    return True
+
+
+def main() -> None:
+    """
+    main
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "starting_path", help="path from which to recur downward looking for zarr dirs"
+    )
+    parser.add_argument(
+        "--proto", help="prototype for zarr search. Remember to quote it!",
+        default="**/*anndata.zarr"
+    )
+    parser.add_argument(
+        "--dryrun",
+        help="describe the steps that would be taken but" " do not make changes",
+        action="store_true",
+    )
+
+    args = parser.parse_args()
+
+    dryrun = args.dryrun
+
+    start_path = Path(args.starting_path)
+    if not start_path.is_dir():
```

Comment:
Can we add a check to make sure its not operating from the root directory as a starting point